pkg/tagset/hashing_tags_accumulator.go

```diff
@@ -19,6 +20,35 @@ type HashingTagsAccumulator struct {
 	hashedTags
 }
 
+// tagName extracts the tag name portion from the tag.
+func tagName(tag string) string {
+	tagNamePos := strings.Index(tag, ":")
+	if tagNamePos < 0 {
+		tagNamePos = len(tag)
+	}
+
+	return tag[:tagNamePos]
+}
+
+type tagMatcher interface {
+	KeepTag(string) bool
+}
+
+// FilterTags removes configured tags and their hashes from the accumulated tags.
+func (h *HashingTagsAccumulator) FilterTags(tagMatcher tagMatcher) {
```

Comment:
For singular interfaces, it is more idiomatic to use a function:

```suggestion
func (h *HashingTagsAccumulator) FilterTags(func (string) bool) {
```

See https://pkg.go.dev/slices#DeleteFunc